Sources/MessagingPush/MessagingPushImplementation.swift

```diff
@@ -228,28 +226,47 @@ internal class MessagingPushImplementation: MessagingPushInstance {
         }
 
         // Time to handle rich push notifications.
-        guard let pushContent = PushContent.parse(notificationContent: response.notification.request.content,
-                                                  jsonAdapter: jsonAdapter)
+        guard let pushContent = CustomerIOParsedPushPayload
+            .parse(notificationContent: response.notification.request.content,
+                   jsonAdapter: jsonAdapter)
         else {
             // push does not contain a CIO rich payload, so end early
-            return false
+            return nil
         }
 
         cleanup(pushContent: pushContent)
```

Comment:
Cleanup's job is to delete content that was used for *displaying* the push. Since this function `userNotificationCenter(didReceive)` is called after a push is interacted with, we can cleanup, yes. 

I'll add a comment to the code though because that cleanup function's job is focused on just cleaning up after a push is clicked on. We should be careful not to do  too much with that cleanup function.